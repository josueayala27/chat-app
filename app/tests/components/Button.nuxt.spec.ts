import { BaseButton } from '#components'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

describe('baseButton', () => {
  it('renders the default button correctly (snapshot)', async () => {
    const wrapper = await mountSuspended(BaseButton, {
      props: {
        content: 'Click me',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with content', async () => {
    const wrapper = await mountSuspended(BaseButton, {
      props: {
        content: 'Click me',
      },
    })

    expect(wrapper.text()).toContain('Click me')
  })

  it('renders slot content when not loading', async () => {
    const wrapper = await mountSuspended(BaseButton, {
      slots: {
        default: 'Slot Content',
      },
    })

    expect(wrapper.text()).toContain('Slot Content')
  })

  it('renders loading icon when loading is true', async () => {
    const wrapper = await mountSuspended(BaseButton, {
      props: {
        loading: true,
      },
    })

    const icon = wrapper.find('span')
    expect(icon.exists()).toBe(true)
  })

  it('applies custom UI class via props', async () => {
    const wrapper = await mountSuspended(BaseButton, {
      props: {
        ui: {
          base: 'bg-red-500',
        },
      },
    })

    const btn = wrapper.find('button')
    expect(btn.attributes('class')).toContain('bg-red-500')
  })
})
