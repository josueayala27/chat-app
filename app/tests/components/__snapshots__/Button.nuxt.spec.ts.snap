// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button component > renders the default button correctly (snapshot) 1`] = `"<button class="bg-sky-500 hover:bg-sky-600 rounded-lg h-9 px-4 flex items-center justify-center text-white cursor-pointer text-sm duration-200">Click me</button>"`;
